### PR TITLE
Move functions into trait.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Move functions in trait. 

## Relevant Technical Choices

These three functions are used in a couple of places. It is pretty simple to move them into a trait. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

**Comment by swissspidy** [comment]
These seem better suited to be in their own traits or classes. When I want to access field state information I don‘t want to use the script data trait.

File: includes/functions.php [hunk 1]
```diff
@@ -72,117 +66,3 @@ function render_theme_stories() {
...
- *
- * @since 1.5.0
- * @access private
- * @todo Move out of this file
- *
- * @param string $view View Type.
- *
- * @return GridView|ListView|CircleView|CarouselView
- */
-function get_field( $view = 'grid' ) {
```
Comment (swissspidy) [inline includes/functions.php:87]: This is probably better suited as a method in a FieldState factory or something.

**Comment by spacedmonkey** [comment]
> These seem better suited to be in their own traits or classes. When I want to access field state information I don‘t want to use the script data trait.

That would mean that a trait would need to use another trait. Is that possible? If so, yeah, it can be it's own trait.

File: includes/functions.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/functions.php:87]: Should I create a new FieldState factory? This would only be used for single use that `get_field` would be used in.

**Comment by swissspidy** [comment]
> > These seem better suited to be in their own traits or classes. When I want to access field state information I don‘t want to use the script data trait.
> 
> That would mean that a trait would need to use another trait. Is that possible? If so, yeah, it can be it's own trait.

That's not what I meant, but I didn't have a specific solution in mind. I just don't think this change is much of an improvement as-is and that it needs some thinking first on how to structure this in a better way.

When looking at all these traits and functions I see a need for reorganizing it in a bigger way than just moving everything into this one trait here. Heck, I'd probably even remove `Stories_Script_Data` entirely and create a dedicated script for loading that data which is a dependency for the TinyMCE & customizer scripts.

NB, this is why we should really look into a more holistic refactoring of the whole PHP codebase architecture. (#5359 comes to mind)

**Review by swissspidy — APPROVED** [review]
Approving it now regardless, because I don't care about this part too much. But this really needs fixing.

File: includes/functions.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/functions.php:87]: Maked to a factory now.

**Comment by spacedmonkey** [comment]
For what is worth, I don't love the way it is now, but it is an improvement to what it was, with random functions that I was worried developers might use.

**Comment by swissspidy** [comment]
Now we have just a bunch of classes and traits and factories developers might use 🤷

**Pull request merged**